libs/utils/android/workloads/viewer.py

```diff
@@ -0,0 +1,112 @@
+# SPDX-License-Identifier: Apache-2.0
+#
+# Copyright (C) 2017, ARM Limited and contributors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License"); you may
+# not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+import re
+import os
+import logging
+
+from time import sleep
+
+from android import Screen, System
+from android.workload import Workload
+
+
+class Viewer(Workload):
+    """
+    Android generic Viewer workload
+    """
+    
+    # Let the system pick the best package
+    package = ''
+
+    def __init__(self, test_env):
+        super(Viewer, self).__init__(test_env)
+
+        # Set of output data reported by the viewer
+        self.db_file = None
+
+    def pre_viewing(self):
```

Comment:
Also for your below comment on the gfxinfo_reset, I can't reply directly to it so I'll do it here: as there are other workloads that use gfxinfo and then call sleep, I was thinking of writing a separate pull request that removes all of those calls and adds it to the System call itself.